src/google/protobuf/stubs/common.h

```diff
@@ -47,6 +47,10 @@
 #include "google/protobuf/stubs/platform_macros.h"
 #include "google/protobuf/stubs/port.h"
 
+#if !defined(PROTOBUF_FORCE_OLD_CPP_VERSION_WE_KNOW_WHAT_WE_ARE_DOING) && __cplusplus < 201402L
```

Comment:
Can you remove this guard?  Our code is completely broken pre-C++14 right now, so if someone were to want to use it they'd need to make changes anyway (i.e. they could just delete this check)

Reply:
I see, that's fair. I wanted something to emulate this as a lower-bound guard but if it already explodes below C++14 this is somewhat redundant.

Reply:
Sorry I think you misunderstood.  I love this C++14 guard, it'll give a very clear error message instead of some random compiler error.  The issue I had was just with your `PROTOBUF_FORCE_OLD_CPP_VERSION_WE_KNOW_WHAT_WE_ARE_DOING` short-circuit, which seems unnecessary

Reply:
i see! i wanted to offer a pre-built way to bypass the error without patching the header files; like if someone is on `__cplusplus == 201103L` but has `<string_view>` and is defining `ABSL_USES_STD_STRING_VIEW` etc., but i don't know what all possibilities exist. If it's messy enough to do this we can drop it.

Reply:
Yea the issue is that we make no guarantees that C++11 will work, and in fact I'm pretty confident it doesn't.  So in this case they'd need hack our code anyway to get it compiling, and could just remove this guard then